### PR TITLE
added new version vehicle_positions

### PR DESCRIPTION
**Pull request by julyto**
I created a new version for the position. I changed the name of the API to "vehicle_positions"
I will delete the old one after validation.

### CODE REVIEW

**Review by kadhikari — CHANGES_REQUESTED** [review]

File: cmd/forseti/main.go [hunk 1]
```diff
@@ -201,7 +201,7 @@ func main() {
 	VehicleOccupancies(manager, &config, router, location)
 
 	// With vehicle locations
```
[the remark this answers is not in the document]
Comment (julyto) [reply cmd/forseti/main.go:203]: Changes are made in the main

File: api/api.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ type StatusResponse struct {
 	LastEquipmentUpdate time.Time     `json:"last_equipment_update"`
 	FreeFloatings       LoadingStatus `json:"free_floatings,omitempty"`
 	VehicleOccupancies  LoadingStatus `json:"vehicle_occupancies,omitempty"`
-	VehicleLocations    LoadingStatus `json:"vehicle_locations,omitempty"`
+	VehiclePositions    LoadingStatus `json:"vehicle_locations,omitempty"`
```
[the remark this answers is not in the document]
Comment (julyto) [reply api/api.go:40]: Yes I changed

File: cmd/forseti/main.go [hunk 3]
```diff
@@ -55,24 +55,24 @@ type Config struct {
...
+	PositionsFilesURI      url.URL
+	PositionsNavitiaURIStr string `mapstructure:"locations-navitia-uri"`
+	PositionsNavitiaURI    url.URL
+	PositionsServiceURIStr string `mapstructure:"locations-service-uri"`
+	PositionsServiceURI    url.URL
+	PositionsNavitiaToken  string        `mapstructure:"locations-navitia-token"`
+	PositionsServiceToken  string        `mapstructure:"locations-service-token"`
+	PositionsRefresh       time.Duration `mapstructure:"locations-refresh"`
+	PositionsCleanVJ       time.Duration `mapstructure:"locations-clean-vj"`
+	PositionsCleanVP       time.Duration `mapstructure:"locations-clean-vp"`
```
[the remark this answers is not in the document]
Comment (julyto) [reply cmd/forseti/main.go:69]: Yes

File: internal/vehiclepositions/prometheus.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package vehiclepositions
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	VehiclePositionsLoadingDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+		Namespace: "forseti",
+		Subsystem: "vehicle_locations",
```
Comment (azime) [inline internal/vehiclepositions/prometheus.go:10]:
```suggestion
		Subsystem: "vehicle_positions",
```
No ?

File: internal/vehiclepositions/prometheus.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+		Name:      "load_durations_seconds",
+		Help:      "http request latency distributions.",
+		Buckets:   prometheus.ExponentialBuckets(0.001, 1.5, 15),
+	})
+
+	VehiclePositionsLoadingErrors = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: "forseti",
+		Subsystem: "vehicle_locations",
```
Comment (azime) [inline internal/vehiclepositions/prometheus.go:18]:
```suggestion
		Subsystem: "vehicle_positions",
```

File: internal/vehiclepositions/vehiclepositions_test.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_NewVehiclePosition(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+	location, err := time.LoadLocation("Europe/Paris")
+	require.Nil(err)
+	date, err := time.ParseInLocation("2006-01-02", "2021-02-22", location)
```
Comment (azime) [inline internal/vehiclepositions/vehiclepositions_test.go:17]: Datetime, no ?

File: internal/vehiclepositions/prometheus.go (hunk 4 above)
Comment (julyto) [reply internal/vehiclepositions/prometheus.go:10]: :+1:

File: internal/vehiclepositions/prometheus.go (hunk 5 above)
Comment (julyto) [reply internal/vehiclepositions/prometheus.go:18]: :+1:

File: internal/vehiclepositions/vehiclepositions_test.go (hunk 6 above)
Comment (julyto) [reply internal/vehiclepositions/vehiclepositions_test.go:17]: As for Position, the API only uses a date "20060102"

**Review by azime — APPROVED** [review]

File: internal/vehiclepositions/vehiclepositions.go [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+
+func (d *VehiclePositions) CleanListVehiclePositions() {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	if d.vehiclePositions != nil {
+		for k := range d.vehiclePositions {
+			delete(d.vehiclePositions, k)
+		}
+	}
```
[the remark this answers is not in the document]
Comment (julyto) [reply internal/vehiclepositions/vehiclepositions.go:37]: :+1:

**Review by benoit-bst — APPROVED** [review]

**Review by kadhikari — APPROVED** [review]

**Pull request merged**